From 9383f16d3856f5b66a05e1df6866d0aff5cf413f Mon Sep 17 00:00:00 2001
From: Pauli Nieminen <suokkos@gmail.com>
Date: Mon, 1 Mar 2010 11:37:11 +0200
Patch-mainline: 2.6.34
References: fate#310916
Git-commit: 55a5cb5d594c18b3147a2288b00ea359c1a38cf8
Subject: [PATCH 0477/2588] drm/radeon: Fix printf type warning in 64bit
 system.

Type of iterator was promoted to unsigned long in 64bit systems.

*header is small structure so it is alwas safe to cast return value
of sizeof operator to int.

Signed-off-by: Pauli Nieminen <suokkos@gmail.com>
(cherry picked from commit 55a5cb5d594c18b3147a2288b00ea359c1a38cf8)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/radeon/r300_cmdbuf.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/radeon/r300_cmdbuf.c b/drivers/gpu/drm/radeon/r300_cmdbuf.c
index 7f59352..ea46d55 100644
--- a/drivers/gpu/drm/radeon/r300_cmdbuf.c
+++ b/drivers/gpu/drm/radeon/r300_cmdbuf.c
@@ -767,7 +767,7 @@ static __inline__ int r300_emit_packet3(drm_radeon_private_t *dev_priv,
 		default:
 			DRM_ERROR("bad packet3 type %i at byte %d\n",
 				  header.packet3.packet,
-				  cmdbuf->buffer->iterator - sizeof(header));
+				  cmdbuf->buffer->iterator - (int)sizeof(header));
 			return -EINVAL;
 		}
 
@@ -1153,7 +1153,7 @@ int r300_do_cp_cmdbuf(struct drm_device *dev,
 		default:
 			DRM_ERROR("bad cmd_type %i at byte %d\n",
 				  header->header.cmd_type,
-				  cmdbuf->buffer->iterator - sizeof(*header));
+				  cmdbuf->buffer->iterator - (int)sizeof(*header));
 			ret = -EINVAL;
 			goto cleanup;
 		}
-- 
1.7.6

